src/main/java/radiounit/ConcreteRadioUnit.java

```diff
@@ -0,0 +1,191 @@
+package radiounit;
+
+import java.util.List;
+
+import common.AlarmStatusLevel;
+import common.Carrier;
+import common.FrequencyBand;
+import common.RatType;
+import common.Vendor;
+import radiounit.radiostate.*;
+
+/**
+ * 
+ * @author esiumat
+ *
+ */
+public class ConcreteRadioUnit extends AbstractRadioUnit {
+
+	/**
+	 * Constructor for class
+	 */
+	public ConcreteRadioUnit(String ipAddress, String name, Vendor vendor, RatType ratType) {
+		this.ipAddress = ipAddress;
+		this.name = name;
+		this.vendor = vendor;
+		this.ratType = ratType;
+		this.alarmStatus = AlarmStatusLevel.NO_ALARM;
+		this.state = new IdleState(this);
+		// TODO: instantiate a RadioCommandExecutor
+		// TODO: get RadioCommandExecutorFactory instance here.
+
+	}
+
+	/**
+	 * Setup radio
+	 */
+	@Override
+	public void setup() {
+		cmdExecutor.setup();
```

Comment:
Minor (future) - we could add some (optional?) print statements in each of these statements to show flow of execution